dashboard/src/app/projects/create-project/create-project.controller.js

```diff
@@ -570,23 +570,42 @@ export class CreateProjectCtrl {
         if (estimateTypes.length > 0) {
           // wait estimate are all finished
           let waitEstimate = this.$q.all(estimatePromises);
+          let attributesByMatchingType = new Map();
 
           waitEstimate.then(() => {
-            var firstMatchingType;
-            var firstMatchingResult;
+            let firstMatchingType;
             estimateTypes.forEach((type) => {
               let resultEstimate = projectService.getEstimate(projectName, type);
               // add attributes
-              // there is a matching estimate
-              if (Object.keys(resultEstimate.attributes).length > 0 && 'java' !== type && !firstMatchingType) {
-                firstMatchingType = type;
-                firstMatchingResult = resultEstimate.attributes;
+              if (Object.keys(resultEstimate.attributes).length > 0) {
+                attributesByMatchingType.set(type, resultEstimate.attributes);
+              }
+            });
+            
+            attributesByMatchingType.forEach((attributes, type) => {
+              if (!firstMatchingType) {
+                let projectType = projectTypesByCategory.get(type);
+                if (projectType && projectType.parents) {
+                  projectType.parents.forEach((parentType) => {
+                    if (parentType === 'java') {
```

Comment:
When I try to get project types:     /wsagent/ext/project-type (GET)
Response:
0: {persisted: false, parents: ["java"], mixable: false, primaryable: true,…id: "javac"...}
1: {persisted: false, parents: ["blank"], mixable: false, primaryable: true,…id: "cpp"...}
2: {persisted: false, parents: ["blank"], mixable: false, primaryable: true, ... id: "python",…}
3: {persisted: false, parents: [], mixable: false, primaryable: true, ancestors: [], id: "blank",…}
4: {persisted: false, parents: ["blank"], mixable: false, primaryable: true, ... id: "java",…}
5: {persisted: false, parents: ["blank"], mixable: true, primaryable: false, ... id: "git",…}
6: {persisted: false, parents: ["blank"], mixable: false, primaryable: true, ... id: "node-js",…}
7: {persisted: false, parents: ["blank"], mixable: false, primaryable: true, ... id: "c",…}
8: {persisted: false, parents: ["blank"], mixable: true, primaryable: false, ... id: "subversion",…}
9: {persisted: false, parents: ["java"], mixable: false, primaryable: true, ... id: "maven",  ...}

@vparfonov: What should we do for 4.4 version with this parent type in case when we have more than one valid type with attributes( like 'maven' and 'jjavac')?